browserdebuggertools/chrome/interface.py

```diff
@@ -21,38 +22,83 @@ class ChromeInterface(object):
             interface.navigate(url="https://github.com/scivisum/browser-debugger-tools")
     """
 
-    def __init__(self, port, host="localhost", timeout=30, domains=None):
+    def __init__(
+        self,
+        port: int,
+        host: str = "localhost",
+        timeout: int = 30,
+        domains: Optional[dict] = None,
+        attach: bool = True
+    ):
         """ Initialises the interface starting the websocket connection and enabling
             a series of domains.
 
         :param port: remote-debugging-port to connect.
+        :param host: host where the browser is
         :param timeout: Timeout between executing a command and receiving a result.
         :param domains: Dictionary of dictionaries where the Key is the domain string and the Value
-        is a dictionary of the arguments passed with the domain upon enabling.
+            is a dictionary of the arguments passed with the domain upon enabling.
+        :param attach: If set to true, the interface will attach to the first page target found.
+            If there are no  page targets, a new tab will be created.
+        """
+        self._targets_manager = TargetsManager(timeout, port, host=host, domains=domains)
+        if attach:
+            self.switch_target()
+
+    @property
+    def targets(self):
+        return self._targets_manager.refresh_targets()
+
+    def switch_target(self, target_id=None):
+        """
+        Switches the current target to the one specified by target_id. If no target_id is
+        specified, the first page target found will be used. If no page targets are found, a new
+        tab will be created.
         """
-        # type: WSSessionManager
-        self._session_manager = WSSessionManager(port, host, timeout, domains=domains)
-        self._dom_manager = _DOMManager(self._session_manager)
+        self._targets_manager.refresh_targets()
+        if target_id:
+            if target_id in self._targets_manager.targets:
+                self._targets_manager.switch_target(target_id)
+            else:
+                raise TargetNotFoundError(f"Target with id {target_id} not found")
+        else:
+            page_targets = [
+                target_id for target_id, target in self._targets_manager.targets.items()
+                if target.type == "page"
+            ]
+            if self._targets_manager.targets:
```

Comment:
Actually self._targets_manager is an instance it can't be empty.
What we really want is to just know if we have a page target to attach to, and we already know that in page_targets. So we should just condition on that.

Reply:
Done